diff --git a/node_modules/node-firebird/lib/wire/connection.js b/node_modules/node-firebird/lib/wire/connection.js
index 290e4f9..8cad0f2 100644
--- a/node_modules/node-firebird/lib/wire/connection.js
+++ b/node_modules/node-firebird/lib/wire/connection.js
@@ -158,8 +158,8 @@ class Connection {
                         xdr.pos = 0;
                         self._xdr = xdr;
     
-                        if (self.accept.protocolMinimumType === Const.ptype_lazy_send && self._queue.length > 0) {
-                            self._queue[0].lazy_count = 2;
+                        if (self.accept && self.accept.protocolMinimumType === Const.ptype_lazy_send && self._queue.length > 0) {
+                            if (self._queue[0]) self._queue[0].lazy_count = 2;
                         }
                         return;
                     }
